### ISSUE TITLE
two-way coupling

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/vector-addition#234]
Related to code review #231.

> - [ ]  Is there any unnecessary coupling? (e.g., by passing large objects to constructors, or exposing unnecessary properties/functions)

This sim doesn't pass large objects or expose unnecessary properties/functions. But what it does do is create undesirable 2-way (aka circular) relationships.   Specifically, `Vector`, `VectorSet`, and `Graph` all have knowledge of each other. 

The undesirable relationships are:

(1) `Vector` has knowledge of its associated `VectorSet` and `Graph`:

```js
  class Vector extends RootVector {

    /**
     * @param {Vector2} initialTailPosition - starting tail position of the vector
     * @param {Vector2} initialComponents - starting components of the vector
     * @param {Graph} graph - the graph the vector belongs to
     * @param {VectorSet} vectorSet - the vector set the vector belongs to
     * @param {string|null} symbol - the symbol for the vector (i.e. 'a', 'b', 'c', ...)
     * @param {Object} [options] - not propagated to super
     */
    constructor( initialTailPosition, initialComponents, graph, vectorSet, symbol, options ) {
```
<br>

(2) `VectorSet` has knowledge of its associated Graph:

```js
  class VectorSet {

    /**
     * @param {Graph} graph - the graph the VectorSet belongs to
     * @param {EnumerationProperty.<ComponentVectorStyles>} componentStyleProperty - component style for all vectors
     * @param {BooleanProperty} sumVisibleProperty - controls whether the sum vector is visible
     * @param {VectorColorPalette} vectorColorPalette - color palette for vectors in this set
     * @param {Object} [options]
     */
    constructor( graph, componentStyleProperty, sumVisibleProperty, vectorColorPalette, options ) {
```
<br>

This type of 2-way coupling makes it difficult to change things. In general, we want to keep general things (e.g. `Vector`) from having knowledge of the things that use/contain them (e.g. `VectorSet`).

**Comment by pixelzoom** [comment]
I discussed this issue several weeks ago with @ariel-phet, when I first started running into changes that were difficult/costly because of coupling.  Eliminating this coupling would require fundamental systemic changes, would be costly (my WAG was ~60 hours) and would undoubtedly be destabilizing -- not the kind of change we want to make this late in the game, when the sim is polished and (currently) has no known functional problems.   So the decision was to leave things "as is", and incur the cost during future maintenance/enhancement.
